crates/transaction-pool/src/lib.rs

```diff
@@ -473,6 +473,18 @@ where
         self.pool.get_transactions_by_sender(sender)
     }
 
+    fn get_transactions_by_sender_and_nonce(
+        &self,
+        sender: Address,
+        nonce: u64,
+    ) -> Vec<Arc<ValidPoolTransaction<Self::Transaction>>> {
```

Comment:
while this can be solved with a default impl using the existing get_transactions_by_sender impl, the more efficient way would be to impl it directly so we can get rid of the vec alloc